runtime/src/master.rs

```diff
@@ -169,12 +169,15 @@ decl_module! {
         /// Execute a proposal that has received enough votes. The proposal is a serialized Call.
         /// This function can be called by anyone, even someone who is not a member of Master.
         ///
-        /// After a sucessful execution, the round number is increased.
+        /// After a successful execution, the round number is increased.
         // TODO: benchmark worst case cost to verify a signature and add it to weight
         #[
-            weight = 10_000
+            weight = (10_000
                 + proposal.get_dispatch_info().weight
-                + T::DbWeight::get().reads(auth.len() as u64)
+                + T::DbWeight::get().reads(auth.len() as u64),
+             proposal.get_dispatch_info().class,
+             proposal.get_dispatch_info().pays_fee,
```

Comment:
I think we should use the new system for returning `Pays::No` here instead of inheriting transaction cost from the contained transaction.

Reply:
"cost" is a poor word choice. I mean pay status e.g. `Pays::No` or `Pays::yes`.

Reply:
The objective was to keep the corresponding call be responsible for its payment details and master's just respects that

Reply:
What are the upsides to that?